src/modules/category/adapter/controller/category.controller.ts

```diff
@@ -0,0 +1,44 @@
+import { Body, Controller, Delete, Get, Param, Post, UnprocessableEntityException, UseGuards } from '@nestjs/common';
+import { CategoryService } from '../../usecase/service/category.service';
+import { SupabaseGuard } from 'src/modules/auth/guard/auth.guard';
+import { CategoryInput } from '../dto/category.input';
+import { CategoryOutput } from '../dto/category.output';
+import { ENABLE_CATEGORY_MUTATION } from 'src/modules/common/environment';
+
+@Controller('/categories')
+export class CategoryController {
+    constructor(private readonly categoryService: CategoryService) {}
+
+    @Post()
+    @UseGuards(SupabaseGuard)
+    async createCategory(@Body() category: CategoryInput): Promise<void> {
+        if (!ENABLE_CATEGORY_MUTATION) throw new UnprocessableEntityException('Category mutation is currently disabled');
```

Comment:
@PCalape Can we make the API return a 404 response when this is still disabled?

Reply:
Sure @arnellebalane will do

Reply:
Applied the changes @arnellebalane for your review thanks!